src/Provider/Pelias/Pelias.php

```diff
@@ -0,0 +1,251 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Geocoder package.
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @license    MIT License
+ */
+
+namespace Geocoder\Provider\Pelias;
+
+use Geocoder\Collection;
+use Geocoder\Exception\InvalidCredentials;
+use Geocoder\Exception\QuotaExceeded;
+use Geocoder\Exception\UnsupportedOperation;
+use Geocoder\Model\Address;
+use Geocoder\Model\AddressCollection;
+use Geocoder\Query\GeocodeQuery;
+use Geocoder\Query\ReverseQuery;
+use Geocoder\Http\Provider\AbstractHttpProvider;
+use Geocoder\Provider\Provider;
+use Http\Client\HttpClient;
+
+class Pelias extends AbstractHttpProvider implements Provider
+{
+    /**
+     * @var string
+     */
+    private $root;
+
+    /**
+     * @var int
+     */
+    private $version;
+
+    /**
+     * @param HttpClient $client  an HTTP adapter
+     * @param string     $root    url of Pelias API
+     * @param int        $version version of Pelias API
+     */
+    public function __construct(HttpClient $client, string $root, int $version = 1)
+    {
+        $this->root = sprintf('%s/v%d', rtrim($root, '/'), $version);
+        $this->version = $version;
+
+        parent::__construct($client);
+    }
+
+    /**
+     * @param GeocodeQuery $query
+     * @param array        $query_data Additional query data (API key for instance).
+     *
+     * @return string
+     *
+     * @throws \Geocoder\Exception\Exception
+     */
+    protected function getGeocodeQueryUrl(GeocodeQuery $query, array $query_data = []): string
+    {
+        $address = $query->getText();
+
+        // This API doesn't handle IPs
+        if (filter_var($address, FILTER_VALIDATE_IP)) {
+            throw new UnsupportedOperation('The Pelias provider does not support IP addresses, only street addresses.');
```

Comment:
Minor, but not sure if you want to get the provider name of the current provider here (vs the base provider).

Reply:
Good idea 👍